s3transfer/manager.py

```diff
@@ -27,7 +28,9 @@ def __init__(self,
                  multipart_threshold=8 * MB,
                  max_concurrency=10,
                  multipart_chunksize=8 * MB,
-                 max_queue_size=0):
+                 max_queue_size=0,
```

Comment:
Given each job will be 8MB, I think we should have the default be to cap the max queue size depending on what we feel comfortable as a max memory usage.